mesa_geo/visualization/MapModule.py

```diff
@@ -19,11 +20,11 @@ def __init__(
         self.js_code = "elements.push(" + new_element + ");"
 
     def render(self, model):
-        featurecollection = dict(type="FeatureCollection", features=[])
-        for _, agent in enumerate(model.grid.agents):
-            shape = agent.__geo_interface__()
-            portrayal = self.portrayal_method(agent)
-            for key, value in portrayal.items():
-                shape["properties"][key] = value
-                featurecollection["features"].append(shape)
-        return featurecollection
+        feature_collection = {"type": "FeatureCollection",
+                              "features": [
+                                  {"type": "Feature",
+                                   "geometry": mapping(transform(model.space.Transformer.transform, agent.shape)),
```

Comment:
Would it make sense for `mapping(transform(model.space.Transformer.transform, agent.shape))` to be wrapped as a method of `GeoAgent`, maybe `get_geometry`?

Reply:
This is a good point, although `agent.get_geometry()` will probably just return `agent.shape`. `model.space.Transformer` then transforms agents' coordinates to be consistent with Leaflet's default `epsg:4326` crs. (But why does `GeoSpace` know the default crs of `MapModule` and do the transformation?)

The current crs settings in mesa-geo are a bit convoluted and inconsistent (for example https://github.com/projectmesa/mesa-geo/issues/20). I'll raise an issue regarding this.

Reply:
`mapping(transform(model.space.Transformer.transform, agent.shape))` is way too long, and so I think it would be helpful to package it in a method. Maybe call it instead `agent.get_transformed_geometry`?

Reply:
Yup this looks better. Added the new function.

This makes me wonder whether `agent.shape` should be renamed to `agent.geometry` to be consistent with the GeoDataFrame (as mentioned in https://github.com/projectmesa/mesa-geo/issues/37). But we can do this in a separate PR.